### PR TITLE
Port to Typescript

### PR DESCRIPTION
**Pull request by zone117x**
This lib has as least a couple dependents that are now Typescript -- by porting this those projects no longer have to force import this lib. 

Zero changes to src and test functionality. 

The CDN (standalone dist) bundle may have changes; using webpack instead of browserify -- better treeshaking results in nearly a 50% decrease in file size. 

Changes:
* Removed Flow and replaced with Typescript.
* Replaced the `dist` bundle build with Webpack as opposed to Browserify -- better babel and typescript support, better dead code elimination. 
* Created CircleCI and Codecov.io configs for testing and coverage.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,19 +1,19 @@
 {
   "name": "c32check",
-  "version": "0.0.6",
+  "version": "1.0.0",
   "description": "Crockford base-32 checksum encoding",
   "main": "lib/index",
+  "unpkg": "dist/c32check.js",
+  "jsdelivr": "dist/c32check.js",
```
Comment (kyranjamie) [inline package.json]: Can we add `prettier-config` ?

Comment (kyranjamie) [reply package.json]: `@blockstack/prettierconfig@0.0.4` has trailing comma

Comment (zone117x) [reply package.json]: Added, but did not format the files. It changes most lines so I'm thinking another PR for that

**Comment by zone117x** [comment]
@jcnelson 
> LGTM as long as all of the (extensive) tests pass. Thanks for porting it over!

Just got testing on circleCI up. All passing, line coverage at >99%. The `BIG_DATA_TESTS=1` tests pass as well.

**Pull request merged**